libyul/AsmPrinter.cpp

```diff
@@ -119,12 +117,13 @@ string AsmPrinter::operator()(FunctionDefinition const& _functionDefinition) con
 		);
 	}
 
-	return out + "\n" + (*this)(_functionDefinition.body);
+	return formatSourceLocationComment(_functionDefinition.debugData) + out + "\n" + (*this)(_functionDefinition.body);
 }
 
 string AsmPrinter::operator()(FunctionCall const& _functionCall) const
 {
 	return
+		formatSourceLocationComment(_functionCall.debugData) +
```

Comment:
Please don't rely on the evaluation order here. `formatSourceLocationComment` has to be called before the visit to the arguments, otherwise the `m_lastLocation` logic is wrong.